tests/core/models/answersconfigbuilder.js

```diff
@@ -0,0 +1,131 @@
+  });
+
+  it('throws an error when the experienceKey key is missing from the raw answers config', () => {
+    const config = {
+      apiKey: key
+    };
+    const buildingAnswersConfig = () => new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(buildingAnswersConfig).toThrow(AnswersConfigError);
+    expect(buildingAnswersConfig).toThrow('Missing required `experienceKey`. Type must be {string}');
+  });
+
+  it('throws an error when onStateChange is not a function', () => {
+    const config = {
+      apiKey: key,
+      experienceKey: key,
+      onStateChange: 'not a function'
+    };
+    const buildingAnswersConfig = () => new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(buildingAnswersConfig).toThrow(AnswersConfigError);
+    expect(buildingAnswersConfig).toThrow('onStateChange must be a function. Current type is: string');
+  });
+
+  it('throws an error when onVerticalSearch is not a function', () => {
+    const config = {
+      apiKey: key,
+      experienceKey: key,
+      onVerticalSearch: 'not a function'
+    };
+    const buildingAnswersConfig = () => new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(buildingAnswersConfig).toThrow(AnswersConfigError);
+    expect(buildingAnswersConfig).toThrow('onVerticalSearch must be a function. Current type is: string');
+  });
+
+  it('throws an error when onUniversalSearch is not a function', () => {
+    const config = {
+      apiKey: key,
+      experienceKey: key,
+      onUniversalSearch: 'not a function'
+    };
+    const buildingAnswersConfig = () => new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(buildingAnswersConfig).toThrow(AnswersConfigError);
+    expect(buildingAnswersConfig).toThrow('onUniversalSearch must be a function. Current type is: string');
+  });
+
+  it('throws an error when sessionTrackingEnabled is not a boolean', () => {
+    const config = {
+      apiKey: key,
+      experienceKey: key,
+      sessionTrackingEnabled: 'not a boolean'
+    };
+    const buildingAnswersConfig = () => new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(buildingAnswersConfig).toThrow(AnswersConfigError);
+    expect(buildingAnswersConfig).toThrow('sessionTrackingEnabled must be a boolean. Current type is: string');
+  });
+
+  it('properly handles the inclusion of sandboxPrefix in the apiKey', () => {
+    const config = {
+      apiKey: `${SANDBOX}-${key}`,
+      experienceKey: key
+    };
+    const answersConfig = new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(answersConfig.apiKey).toBe(key);
+    expect(answersConfig.environment).toBe(SANDBOX);
+  });
+
+  it('has defaults properly applied', () => {
+    const config = {
+      apiKey: key,
+      experienceKey: key
+    };
+
+    const answersConfig = new AnswersConfigBuilder().setRawConfig(config).build();
+    expect(answersConfig.environment).toBe(PRODUCTION);
```

Comment:
nit: We generally try to minimize the number of expects in each test. For example, you may be able to use `.toMatchObject` to do this in one expect. I don't think you need to change this as it's pretty clear what's going on, but it's something to keep in mind